src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReconciliationService.java

```diff
@@ -0,0 +1,114 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.reports;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ReceivedZipFile;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ReceivedZipFileRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.Discrepancy;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.DiscrepancyType;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.ReconciliationStatement;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.ZipFileData;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toMap;
+import static uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.DiscrepancyType.PAYMENT_DCNS_MISMATCH;
+import static uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.DiscrepancyType.RECEIVED_BUT_NOT_REPORTED;
+import static uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.DiscrepancyType.REPORTED_BUT_NOT_RECEIVED;
+import static uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.DiscrepancyType.SCANNABLE_DOCUMENT_DCNS_MISMATCH;
+
+@Service
+public class ReconciliationService {
+    private final ReceivedZipFileRepository receivedZipFileRepository;
+    private final ReceivedZipFileConverter receivedZipFileConverter;
+
+    public ReconciliationService(
+        ReceivedZipFileRepository receivedZipFileRepository,
+        ReceivedZipFileConverter receivedZipFileConverter
+    ) {
+        this.receivedZipFileRepository = receivedZipFileRepository;
+        this.receivedZipFileConverter = receivedZipFileConverter;
+    }
+
+    public List<Discrepancy> getReconciliationReport(ReconciliationStatement reconciliationStatement) {
+        List<ReceivedZipFile> receivedZipFiles =
+            receivedZipFileRepository.getReceivedZipFilesReportFor(reconciliationStatement.date);
+        List<ZipFileData> receivedZipFileDataList =
+            new ArrayList<>(receivedZipFileConverter.convertReceivedZipFiles(receivedZipFiles));
```

Comment:
is this needed?

Reply:
Yes. The repository returns cartesian product of payment DCNs and scannable document DCNs, and `receivedZipFileConverter` transforms it into 2 proper lists.

Reply:
can it be just
```
List<ReceivedZipFileData> receivedZipFileDataList =
 receivedZipFileConverter.convertReceivedZipFiles(receivedZipFiles));
```
? (no wrapping in `new ArrayList(...)`

Reply:
My bad